optaplanner-examples/src/main/java/org/optaplanner/examples/batchscheduling/domain/solver/RoutePathStrengthWeightFactory.java

```diff
@@ -0,0 +1,26 @@
+package org.optaplanner.examples.batchscheduling.domain.solver;
+
+import java.io.Serializable;
+import java.util.Comparator;
+
+import org.optaplanner.examples.batchscheduling.domain.AllocationPath;
+
+public class RoutePathStrengthWeightFactory implements Comparator<AllocationPath>, Serializable {
+
+    private static final long serialVersionUID = 2970371491784586248L;
+
+    public int compare(AllocationPath a, AllocationPath b) {
+        if ((a != null) && (b != null)) {
+            if (a.getRoutePathList().size() == b.getRoutePathList().size()) {
+                return 0;
+            } else if (a.getRoutePathList().size() > b.getRoutePathList().size()) {
```

Comment:
`size()` on a list could possibly be an expensive operation. 
I suggest you calculate those sizes only once, you could get some nice micro optimization there.